File: leetcode/0819-most-common-word.py
from collections import Counter


class Solution:
    def mostCommonWord(self, paragraph: str, banned: List[str]) -> str:
        for char in ["!", "?", "'", ",", ";", "."]:
            paragraph = paragraph.replace(char, " ")
        banned = set(banned)
        word_count = Counter(
            word.lower()
            for word in paragraph.split(" ")
            if word and word.lower() not in banned
        )

        return word_count.most_common(1)[0][0]

File: src/0819-most-common-word.js


/**
 * @param {string} paragraph
 * @param {string[]} banned
 * @return {string}
 */
var mostCommonWord = function (paragraph, banned) {
    let words = paragraph.replace(/[^\w\s]/gi, ' ').replace(",", " ").split(" ").map((x) => x.toLowerCase().trim()).filter((x) => x != "")
    let set = new Set(banned)
    let obj = {}

    let hightFrecc = ["x", 0]
    for (const word of words) {

        if (obj[word]) {

            obj[word] = obj[word] + 1
        } else if (!set.has(word)) {

            obj[word] = 1
        }
        if (obj[word] && obj[word] > hightFrecc[1]) hightFrecc = [word, obj[word]]
    }
    return hightFrecc[0]
};